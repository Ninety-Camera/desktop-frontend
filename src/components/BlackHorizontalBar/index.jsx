import { findByLabelText } from "@testing-library/react";
import React from "react";
import { Button } from "@mui/material";
import "@fontsource/inter";
// import {useNavigate} from "react-router-dom";

export default function BlackHorizontalBar(props) {
  // const navigate = useNavigate();
  const { phrase } = props;
  return (
    <div
      style={{
        height: 70,
        backgroundColor: "#2F2E41",
        color: "white",
        fontSize: 30,
        fontFamily: "inter",
        fontStyle: "normal",
        fontWeight: 700,
        lineHeight: 48,
        paddingLeft: 50,
        display: "flex",
        alignItems: "center",
        position: "sticky",
        top: "0%",
        width: "100%",
        // justifyContent: "center",
      }}
    >
      {phrase}
      {/* <Button
        variant="contained"
        sx={{
          background: "#6C63FF",
          width: "8%",
          height: "60%",
          align: "right",
        }}
        // onClick={() => navigate("/dashboard")}
      >
        Log out
      </Button> */}
    </div>
  );
}
