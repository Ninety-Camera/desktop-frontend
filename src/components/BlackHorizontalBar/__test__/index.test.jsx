import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import ReactDom from "react-dom";
import BlackHorizontalBar from "..";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import { useState } from "react";

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDom.render(<BlackHorizontalBar />, div);
});

it("renders form correctly", () => {
  //use for props passing
  const { getByTestId } = render(
    <BlackHorizontalBar title="Ninety Camera" buttonText="Register" />
  );
  expect(getByTestId("blackHorizontalBar")).toHaveTextContent("Ninety Camera");
});

it("matches snapshot", () => {
  const tree = renderer
    .create(<BlackHorizontalBar title="Ninety Camera" buttonText="Register" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
