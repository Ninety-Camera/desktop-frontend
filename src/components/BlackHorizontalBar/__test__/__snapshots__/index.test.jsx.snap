// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="MuiBox-root css-i9gxme"
  data-testid="blackHorizontalBar"
>
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-dclvwo-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
        variant="contained"
      >
        Register
      </button>
    </div>
  </header>
</div>
`;
